pkg/runtime/runtime.go

```diff
@@ -308,15 +292,33 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 	}
 	log.Infof("API gRPC server is running on port %v", a.runtimeConfig.APIGRPCPort)
 
+	// Start HTTP Server
+	a.startHTTPServer(a.runtimeConfig.HTTPPort, a.runtimeConfig.ProfilePort, a.runtimeConfig.AllowedOrigins, pipeline)
+	log.Infof("http server is running on port %v", a.runtimeConfig.HTTPPort)
+
 	err = a.startGRPCInternalServer(grpcAPI, a.runtimeConfig.InternalGRPCPort)
 	if err != nil {
 		log.Fatalf("failed to start internal gRPC server: %s", err)
 	}
 	log.Infof("internal gRPC server is running on port %v", a.runtimeConfig.InternalGRPCPort)
 
-	// Start HTTP Server
-	a.startHTTPServer(a.runtimeConfig.HTTPPort, a.runtimeConfig.ProfilePort, a.runtimeConfig.AllowedOrigins, pipeline)
-	log.Infof("http server is running on port %v", a.runtimeConfig.HTTPPort)
+	a.blockUntilAppIsReady()
+
+	err = a.createAppChannel()
+	if err != nil {
+		log.Warnf("failed to open %s channel to app: %s", string(a.runtimeConfig.ApplicationProtocol), err)
+	}
+	a.daprHTTPAPI.Set(a.appChannel)
+	grpcAPI.Set(a.appChannel)
+	a.loadAppConfiguration()
+	a.initDirectMessaging(a.nameResolver)
+	err = a.initActors()
+	if err != nil {
+		log.Warnf("failed to init actors: %s", err)
+	}
+
```

Comment:
Needs to set actor runtime 

a.daprHTTPAPI.SetActorRuntime(a.actor)
grpcAPI.SetActorRuntime(a.actor)